### ISSUE TITLE
Querying for a specific attribute value works, but querying with a placeholder doesn't. Excerpt from my code:

### ISSUE THREAD

**Issue by huahaiy** [juji-io/datalevin#277]
Having the same issue. Querying for a specific attribute value works, but querying with a placeholder doesn't. Excerpt from my code:
```clj
(d/q '[:find ?u . :where [?u :garden.user/name _]] @db/conn)
; => nil
(d/q '[:find ?u . :where [?u :garden.user/name "johnny"]] @db/conn)
; => 2
```

I can also confirm that reverting to version `0.9.8` fixed it for me.

_Originally posted by @JohnnyJayJay in https://github.com/juji-io/datalevin/issues/269#issuecomment-2381295108_

**Comment by huahaiy** [comment]
Cannot seem to be able to reproduce this here. Do you have a test case?

**Comment by JohnnyJayJay** [comment]
> Cannot seem to be able to reproduce this here. @JohnnyJayJay  Do you have a test case?
> 

Yeah, I can try to cook up a minimal example, should have done that from the get go. 

For reference, this is the state of the original code where I encountered this problem: 
https://codeberg.org/johnnyjayjay/everblooming-garden/src/commit/ce8abb653f70126c3ad94649afca4ee54a10a64e/src/everblooming_garden/http.clj#L43

The linked line did not return any results despite there being matching entries in the DB. So in that state I played around with queries and found the behaviour I described in my original comment.

Again, I'll follow up with something more straight forward for reproducing

**Comment by JohnnyJayJay** [comment]
Here's my minimal example for this bug. As was pointed out by others, the key seems to be that a query is executed for the attribute before the first insert occurs.

```clj
(ns dtlv-bug
  (:import
   (java.nio.file Files)
   (java.nio.file.attribute FileAttribute))
  (:require
   [datalevin.core :as d]))

(def schema
  {:username
   {:db/valueType :db.type/string
    :db/unique :db.unique/identity}})

(def conn
  (d/get-conn
   (.. (Files/createTempDirectory "dtlv-test" (make-array FileAttribute 0))
       toAbsolutePath
       toString)
   schema))

(d/q '[:find ?u . :where [?u :username _]] @conn)
;; => nil

(d/transact!
 conn
 [{:username "johnny"}])
;; => ...

(d/q '[:find ?u . :where [?u :username _]] @conn)
;; => nil
(d/q '[:find ?u . :where [?u :username "johnny"]] @conn)
;; => 1
```

**Comment by huahaiy** [comment]
This is already fixed in 0.9.11.

**Comment by JohnnyJayJay** [comment]
Right, I appreciate it. I still gave you a test case since you asked for it.
